aws/resource_aws_service_discovery_private_dns_namespace.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsServiceDiscoveryPrivateDnsNamespace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceDiscoveryPrivateDnsNamespaceCreate,
+		Read:   resourceAwsServiceDiscoveryPrivateDnsNamespaceRead,
+		Delete: resourceAwsServiceDiscoveryPrivateDnsNamespaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vpc": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hosted_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceDiscoveryPrivateDnsNamespaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sdconn
+
+	input := &servicediscovery.CreatePrivateDnsNamespaceInput{
+		Name: aws.String(d.Get("name").(string)),
+		Vpc:  aws.String(d.Get("vpc").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreatePrivateDnsNamespace(input)
+	if err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{servicediscovery.OperationStatusSubmitted, servicediscovery.OperationStatusPending},
+		Target:     []string{servicediscovery.OperationStatusSuccess},
+		Refresh:    servicediscoveryOperationRefreshStatusFunc(conn, *resp.OperationId),
+		Timeout:    5 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	opresp, err := stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*opresp.(*servicediscovery.GetOperationOutput).Operation.Targets["NAMESPACE"])
+	return nil
+}
+
+func resourceAwsServiceDiscoveryPrivateDnsNamespaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sdconn
+
+	input := &servicediscovery.GetNamespaceInput{
+		Id: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetNamespace(input)
+	if err != nil {
+		if isAWSErr(err, servicediscovery.ErrCodeNamespaceNotFound, "") {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("description", resp.Namespace.Description)
+	d.Set("arn", resp.Namespace.Arn)
+	if resp.Namespace.Properties != nil {
+		d.Set("hosted_zone", resp.Namespace.Properties.DnsProperties.HostedZoneId)
+	}
+	return nil
+}
+
+func resourceAwsServiceDiscoveryPrivateDnsNamespaceDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sdconn
+
+	input := &servicediscovery.DeleteNamespaceInput{
+		Id: aws.String(d.Id()),
+	}
+
+	resp, err := conn.DeleteNamespace(input)
+	if err != nil {
+		if isAWSErr(err, servicediscovery.ErrCodeNamespaceNotFound, "") {
+			d.SetId("")
+			return nil
+		}
```

Comment:
This check shouldn't be necessary as we perform refresh prior to destroy and `Read()` would wipe the resource from state if it didn't exist.